src/pocketmine/event/player/PlayerEmoteEvent.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+/*
+ *
+ *  ____            _        _   __  __ _                  __  __ ____
+ * |  _ \ ___   ___| | _____| |_|  \/  (_)_ __   ___      |  \/  |  _ \
+ * | |_) / _ \ / __| |/ / _ \ __| |\/| | | '_ \ / _ \_____| |\/| | |_) |
+ * |  __/ (_) | (__|   <  __/ |_| |  | | | | | |  __/_____| |  | |  __/
+ * |_|   \___/ \___|_|\_\___|\__|_|  |_|_|_| |_|\___|     |_|  |_|_|
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * @author PocketMine Team
+ * @link http://www.pocketmine.net/
+ *
+ *
+*/
+
+declare(strict_types=1);
+
+namespace pocketmine\event\player;
+
+use pocketmine\event\Cancellable;
+use pocketmine\Player;
+use pocketmine\utils\UUID;
+
+/**
+ * Class PlayerEmoteEvent
+ * @package pocketmine\event\player
+ */
+class PlayerEmoteEvent extends PlayerEvent implements Cancellable {
+
+	/** @var UUID */
+	protected $emoteId;
+
+	public function __construct(Player $player, UUID $emoteId) {
+		$this->player = $player;
+		$this->emoteId = $emoteId;
+	}
+
+	public function getEmoteId() : UUID{
+		return $this->emoteId;
+	}
+
+	public function setEmoteId(UUID $emoteId) : void{
```

Comment:
I don't think this is a good idea without any validation.

Reply:
In fact, this method is entirely useless, since your implementation does not respect changes made by plugins.